pkg/errors/errors.go

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+// The idea of writing errors package in thanos is highly motivated from the Tast project of Chromium OS Authors. However, instead of
+// copying the package, we end up writing our own simplified logic borrowing some ideas from the errors and github.com/pkg/errors.
+// A big thanks to all of them.
+
+// Package errors provides basic utilities to manipulate errors with a useful stacktrace. It combines the
+// benefits of errors.New and fmt.Errorf world into a single package.
+package errors
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// base is the fundamental struct that implements the error interface and the acts as the backbone of this errors package.
+type base struct {
+	info  string     // error message passed through calls like errors.Wrap, errors.Errorf
```

Comment:
```suggestion
	info  string     // Error message passed through calls like errors.Wrap, errors.Errorf.
```

Let's use full sentence for all comments, including those short ones, please 🙈

Reply:
Haha, sure.